### PR TITLE
Refactored code

### PR DESCRIPTION
**Pull request by rachelbock**
-Added ApiManager Singleton class for Retrofit. Added url to that class and removed the constant from HomePage. Updated all
instantiations of retrofit to use the Singleton instead.
-Removed NetworkAsyncTask for new user setup. Updated LoginActivity to remove reference to the AsyncTask and to use Retrofitto make post to the database. Added NewUserDataRequest Class to pass the user information to the server
-Updated ClimbsRecyclerViewAdapter to handle project and completed climb checkbox onClick responses within the adapter.
Removed Listener Interfaces from Projects and Climbs Fragments and added onClick responses directly to the adapter.
-Added logic to the NewUserDialogFragment to not allow for blank edit text fields when OK is selected. A Toast will populateif any of the fields have been left blank.
-Removed drawables that are not being used in the app.

### CODE REVIEW

File: app/src/main/java/com/rage/clamber/Activities/LoginActivity.java [hunk 1]
```diff
@@ -121,49 +114,47 @@ public void onFailure(Call<User> call, Throwable t) {
...
         ConnectivityManager connectivityManager = (ConnectivityManager) this.getSystemService(Context.CONNECTIVITY_SERVICE);
         NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
-        if (networkInfo != null && networkInfo.isConnected()) {
-            newUserAsyncTask = new NewUserAsyncTask(this);
-            newUserAsyncTask.execute(user);
-        } else {
-            Toast.makeText(this, R.string.no_internet_connection, Toast.LENGTH_SHORT).show();
-        }
-
+            if (networkInfo != null && networkInfo.isConnected()) {
```
Comment (nonvirtualthunk) [inline app/src/main/java/com/rage/clamber/Activities/LoginActivity.java]: looks like the indentation might be off here

File: app/src/main/java/com/rage/clamber/Activities/LoginActivity.java [hunk 2]
```diff
@@ -121,49 +114,47 @@ public void onFailure(Call<User> call, Throwable t) {
...
+                            startActivity(intent);
+                        }
+                        else if (response.code() == 400){
+                            Toast.makeText(LoginActivity.this, "This username already exists. Create a new username or select 'Existing User'.", Toast.LENGTH_SHORT).show();
+                        }
+                    }
 
-    }
+                @Override
+                public void onFailure(Call<User> call, Throwable t) {
```
Comment (nonvirtualthunk) [inline app/src/main/java/com/rage/clamber/Activities/LoginActivity.java:152]: log on failure

File: app/src/main/java/com/rage/clamber/Networking/ApiManager.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.rage.clamber.Networking;
+
+import retrofit2.Retrofit;
+import retrofit2.converter.jackson.JacksonConverterFactory;
+
+/**
+ * Created by rage on 4/10/16.
```
Comment (nonvirtualthunk) [inline app/src/main/java/com/rage/clamber/Networking/ApiManager.java]: Javadoc!

File: app/src/main/java/com/rage/clamber/Networking/Requests/NewUserDataRequest.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
+package com.rage.clamber.Networking.Requests;
+
+/**
+ * Created by rage on 4/10/16.
```
Comment (nonvirtualthunk) [inline app/src/main/java/com/rage/clamber/Networking/Requests/NewUserDataRequest.java]: More Javadoc!

File: app/src/main/res/layout/fragment_user_info_dialog.xml [hunk 5]
```diff
@@ -3,7 +3,7 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:orientation="vertical"
-    tools:context="com.rage.clamber.Fragments.UserInfoDialogFragment">
+    tools:context=".Fragments.NewUserDialogFragment">
```
Comment (nonvirtualthunk) [inline app/src/main/res/layout/fragment_user_info_dialog.xml]: is this intended to be `.Fragments.NewUserDialogFragment`? Do you not need the whole package path?

**Comment by nonvirtualthunk** [comment]
only minor issues here, good refactoring

File: app/src/main/res/layout/fragment_user_info_dialog.xml (hunk 5 above)
Comment (rachelbock) [inline app/src/main/res/layout/fragment_user_info_dialog.xml]: Fixed

File: app/src/main/java/com/rage/clamber/Activities/LoginActivity.java (hunk 1 above)
Comment (rachelbock) [inline app/src/main/java/com/rage/clamber/Activities/LoginActivity.java]: Fixed

File: app/src/main/java/com/rage/clamber/Activities/LoginActivity.java (hunk 2 above)
Comment (rachelbock) [inline app/src/main/java/com/rage/clamber/Activities/LoginActivity.java:152]: Done

File: app/src/main/java/com/rage/clamber/Networking/ApiManager.java (hunk 3 above)
Comment (rachelbock) [inline app/src/main/java/com/rage/clamber/Networking/ApiManager.java]: Done

File: app/src/main/java/com/rage/clamber/Networking/Requests/NewUserDataRequest.java (hunk 4 above)
Comment (rachelbock) [inline app/src/main/java/com/rage/clamber/Networking/Requests/NewUserDataRequest.java]: Done

**Pull request merged**